Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -180,14 +189,17 @@ def projection_P(self, x, out=None):
     def proximal(self, x, tau, out = None):
         
         ''' Returns the solution of the FGP_TV algorithm '''         
-        self._domain = x.geometry
+        try:
+            self._domain = x.geometry
+        except:
+            self._domain = x
         
         # initialise
         t = 1        
-        tmp_p = self.gradient.range_geometry().allocate(0)  
+        tmp_p = self.gradient.range_geometry().allocate()  
```

Comment:
I would specifically add 0 if you want to initialise to 0, and None if the initial value is not used anyway. Is there a reason to change this and the 2 following lines?

Reply:
The `allocate` method has `value=0` by [default](https://github.com/TomographicImaging/CIL/blob/master/Wrappers/Python/cil/framework/framework.py#L291).

Reply:
yes, but it doesn't hurt to be explicit in-case we change this in the future. Also you have two allocate('none') below that changed to the default, i.e. 0

Reply:
The default case which is `allocate() = allocate(0)` goes to a `.fill` method which is a method in most of the Python-Array libraries, e.g., numpy, cupy. If we use `None` then, an `ImageData` is created and then a `DataContainer` is created using the [`numpy.empty`](https://github.com/TomographicImaging/CIL/blob/master/Wrappers/Python/cil/framework/framework.py#L2764) method  making it restrictive to a Numpy API. In most of my scripts, I use `allocate()`.

Reply:
Changed to `allocate(0)`